### PR TITLE
Experimental SockJS support

### PR DESCRIPTION
**Pull request by kyprizel**
At the moment CodeQL can not work with none of Websocket libraries. This code adds remote flow source for quie popular framework SockJS.

### CODE REVIEW

File: javascript/ql/src/experimental/SockJS/SockJS.qll [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    override string getSourceType() { result = "input from SockJS WebSocket" }
+  }
+
+  /**
+   * Gets a new SockJS server.
+   */
+  private DataFlow::CallNode appCreation() {
+    result = DataFlow::moduleImport("sockjs").getAMemberCall("createServer")
+      or 
+    result = DataFlow::moduleMember("sockjs", "createServer")
```
Comment (erik-krogh) [inline javascript/ql/src/experimental/SockJS/SockJS.qll]:
```suggestion
    result = DataFlow::moduleImport("sockjs").getAMemberCall("createServer")
```

Your line `DataFlow::moduleMember("sockjs", "createServer")` suggests that it is possible to create a SockJS server by running `var server = require("sockjs").createServer`, which is not the case. 

Only the line with `getAMemberCall` is needed.

Comment (kyprizel) [reply javascript/ql/src/experimental/SockJS/SockJS.qll]: Oh, sorry, I used eclipse. Thanks for the suggestion.

File: javascript/ql/src/experimental/SockJS/SockJS.qll [hunk 2]
```diff
@@ -0,0 +1,47 @@
+/**
+ * Provides classes for working with [SockJS](http://sockjs.org).
+ */
+
+import javascript
+import DataFlow::PathGraph
+
```
Comment (erik-krogh) [inline javascript/ql/src/experimental/SockJS/SockJS.qll]:
```suggestion
import javascript

```
You don't need to import `DataFlow::PathGraph` here.  
If you use this library in a path query, that path query should import `DataFlow::PathGraph`.

Comment (erik-krogh) [inline javascript/ql/src/experimental/SockJS/SockJS.qll]:
```suggestion
import javascript

```
You don't need to import `DataFlow::PathGraph` here.  
If you use this library in a path query, that path query should import `DataFlow::PathGraph`.

File: javascript/ql/src/experimental/SockJS/SockJS.qll [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+     * For example:
+     * ```
+     * server.on('connection', function(conn) {
+     *     conn.on('data', function(message) {
+     *       ...
+     *     });
+     *  });
+     * ```
+     */
+    SourceFromSocketJS() {
```
Comment (erik-krogh) [inline javascript/ql/src/experimental/SockJS/SockJS.qll]:
```suggestion
  /**
   * Access to user-controlled data object received from websocket
   * For example:
   * ```
   * server.on('connection', function(conn) {
   *     conn.on('data', function(message) {
   *       ...
   *     });
   *  });
   * ```
   */
  class SourceFromSocketJS extends RemoteFlowSource {
    SourceFromSocketJS() {
```

The QLDoc should be above the class, not above the constructor inside the class.

**Pull request merged**